src/parenthetics.py

```diff
@@ -0,0 +1,21 @@
+"""Determines whether a string has balanced, open, or closed parens."""
+from stack import Stack
+
+
+def eval_parens(input):
+    """Evaluate whether a string has balanced, open, or closed parens."""
+    print('input type:', type(input))
+    if type(input) != str:
+        raise TypeError('Please enter a string.')
+    if '(' not in input and ')' not in input:
```

Comment:
If this is a check if there are any parentheses in the string you might want a line that's more like `if "(" not in input or ")" not in input`